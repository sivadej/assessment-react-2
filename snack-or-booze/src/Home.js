import React from "react";
import { Card, CardBody, CardTitle } from "reactstrap";

function Home({items}) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <h3 className="font-weight-bold">
              Welcome to Silicon Valley's premier dive cafe!
            </h3>
            <h4>Browse our incredibly vast selection of {items.drinks.length} drinks and {items.snacks.length} snacks.</h4>
          </CardTitle>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;
