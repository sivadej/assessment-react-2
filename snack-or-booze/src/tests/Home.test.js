import React from "react";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import Home from "../Home";

it('renders without breaking', ()=> {
  render(<Home itemCounts={{snacks:1, drinks:2}} />);
  render(<Home />);
});

it('matches snapshot', ()=> {
  const {asFragment} = render(<Home itemCounts={{snacks:1, drinks:2}} />);
  expect(asFragment()).toMatchSnapshot();
});

it('displays item counts passed as props', ()=> {
  const { getByText } = render(<Home itemCounts={{snacks:3, drinks:2}} />);
  const subheading = getByText(`2 drinks and 3 snacks`, {exact: false});
  expect(subheading).toBeInTheDocument();
})

it('displays subheading without item counts', () => {
  const { getByText } = render(<Home />);
  const subheading = getByText(`Browse our vast selection of menu items.`);
  expect(subheading).toBeInTheDocument();
})