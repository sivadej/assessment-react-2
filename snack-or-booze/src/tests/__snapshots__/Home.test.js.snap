// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-8"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <h5
          class="card-title"
        />
        <h3
          class="font-weight-bold"
        >
          Welcome to Silicon Valley's premier dive cafe!
        </h3>
        <h4>
          Browse our incredibly vast selection of 2 drinks and 1 snacks.
        </h4>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
